### ISSUE TITLE
No Erase in gw ?!

### ISSUE THREAD

**Issue by gilmod714** [adafruit/Adafruit_Floppy#43]
I built the pi pico GW version and it works OK for read/write with 4.7k resistor on read line  (it's a bit slow, but it works) , however, I noticed that, the "erase" disk function does not work, or not supported ?!

Great work by the way ..

Thank you

**Comment by criscrafter** [comment]
Are you pulling the pin up to 5V or 3.3V?
Sadly I don't have a solution for you, but I can't get mine to work...

**Comment by gilmod714** [comment]
I'm using 3.3v pull .. but I don't think it's that .. it seems to me the "erase" function isn't included in the sketch! .. the read/write works ok though .. but much slower than the stm32 for sure.. I'm using RP2040 zero/and also tried it on pi pico ..

![erase1](https://github.com/user-attachments/assets/58df0e48-306f-47b8-b58a-7543f0766fd1)

![erase2](https://github.com/user-attachments/assets/80d252da-100a-419a-8c8e-a7e8c19d7a21)

![erase3](https://github.com/user-attachments/assets/b85c50b6-032a-491c-ad66-c80b989eb866)

**Comment by jepler** [comment]
EraseFlux (command #17) is not presently supported by our gw-compatible firmware. 

In the official gw firmware, it appears this mode works simply by turning on the write gate and leaving it on for a specified time:
```c

/*
 * ERASE PATH
 */

static uint8_t floppy_erase_prep(const struct gw_erase_flux *ef)
{
    op_delay_wait(DELAY_write);

    if (get_wrprot() == LOW)
        return ACK_WRPROT;

    write_pin(wgate, TRUE);

    floppy_state = ST_erase_flux;
    flux_op.status = ACK_OKAY;
    flux_op.end = time_now() + time_from_samples(ef->ticks);

    return ACK_OKAY;
}

static void floppy_erase(void)
{
    if (time_since(flux_op.end) < 0)
        return;

    floppy_flux_end();

    /* ACK with Status byte. */
    u_buf[0] = flux_op.status;
    floppy_state = ST_command_wait;
    floppy_end_command(u_buf, 1);
}```

I've marked this "help wanted" because I'd be thrilled if someone contributed the implementation.

**Comment by gilmod714** [comment]
It has to erase track by track i think, perhaps not just a delay period..

**Comment by montyleib** [comment]
I believe I've succeeded in adding GW_CMD_ERASE but have no way to test it.  We're in the process of selling our home of 32 years, and we've been packing away all of the non-essentials in preparation for a move.  Needless to say, all of my "old iron" is now boxed up.  :(

@jepler   Jeff - if you can take a look at my fork [https://github.com/montyleib/Adafruit_Floppy/blob/main/examples/greaseweazle/greaseweazle.ino](url) lines 337 to 373, I'd love to get your feedback before I write a commit.

**Comment by gilmod714** [comment]
Awesome work @montyleib .. Thank you for you efforts and wishing you a smooth and easy move :)

**Comment by montyleib** [comment]
@gilmod714 Gilmo - did you try my fork?  I'm on vacation, found a stray brace in my code and committed my change.

**Comment by gilmod714** [comment]
Hello @montyleib ,

I appreciate your work .. sorry I didn't see your previous link before now .. however, the link above takes me to a blank page though .. where can I find your .ino ?! .. so i can test it out and report back

Thank you!


<img width="1242" height="600" alt="Image" src="https://github.com/user-attachments/assets/d6d575d1-d583-492c-b9f4-0f71e8045f9c" />

EDIT: never-mind .. I just found the fork manually .. I'll test it out and provide feedback ;)

**Comment by montyleib** [comment]
Strange, it does the same for me.  Mine on  GitHub is https://github.com/montyleib/Adafruit_Floppy/blob/main/examples/greaseweazle/greaseweazle.ino but I suspect it won't run...  it appears that I failed to re-base some variables copied over from GW.

**Comment by gilmod714** [comment]
there's an issue in line 1331 it seems ..

oh , and I think Jiff has retired and stopped development a while ago (I may have seen that in other posts, or from my other request for WinUae-Pico-gw compatibility)

<img width="1280" height="768" alt="Image" src="https://github.com/user-attachments/assets/703528fb-120d-4bc6-9afc-91f87a80326b" />

**Comment by montyleib** [comment]
greaseweazle.ino only has 729 lines...  and nothing even remotely close to the code in the sketch window.

**Comment by gilmod714** [comment]
oops! .. not sure where i got that from! .. here's the correct one - no compile

<img width="1280" height="768" alt="Image" src="https://github.com/user-attachments/assets/1752b012-db01-42d6-961e-fa310923d75d" />

**Comment by montyleib** [comment]
Yup.  Trying to Python-ize keirf's code isn't going well.  The biggest issue I'm running across is the difference in variable names between Greaseweazle and the AdaFruit code.

That particular section gathers the parameters for which cylinder, head, and number of passes from the received command.  I've found a section of code that does a similar gathering of parameters, I'll work on it some more and let you know.
